### PR TITLE
feat(nms): Expand sidebar on hover

### PR DESCRIPTION
**Pull request by thmsschmitt**
## Summary

First part of #12449. On hover the sidebar is expanded to reveal labels for the sidebar entries. 
The remaining bits of #12449 will be split into several PRs.

## Test Plan
Hover over the sidebar and make sure it is expanded. Click on a sidebar entry and make sure the navigation takes place. Move the mouse away from the sidebar and make sure that the sidebar is collapsed. 
Open the "Account & Settings" popover make sure the sidebar stays expanded. Close the "Account & Settings" popover and make sure the sidebar is collapsed unless you directly hover over the sidebar again.

expandable-sidebar

### CODE REVIEW

File: nms/packages/magmalte/fbc_js_core/ui/stories/mui-components/appSideBar.stories.js [hunk 1]
```diff
@@ -1,76 +0,0 @@
...
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *
- * @flow
- * @format
- */
-
-import AppSideBar from '../../components/layout/AppSideBar';
```
Comment (andreilee) [inline nms/packages/magmalte/fbc_js_core/ui/stories/mui-components/appSideBar.stories.js]: Wow I never even knew there was storybook integration

File: nms/packages/magmalte/fbc_js_core/ui/hooks/useRouter.js [hunk 2]
```diff
@@ -24,16 +24,31 @@ import {__RouterContext as RouterContext} from 'react-router-dom';
...
   const {url} = useRouteMatch();
   return useCallback(
-    (path: string) => (url === '/' && path ? path : `${url}${path}`),
+    (path: string) => {
+      if (!path) {
+        return url;
+      }
+      if (path[0] !== '/') {
+        path = `/${path}`
+      }
```
Comment (andreilee) [inline nms/packages/magmalte/fbc_js_core/ui/hooks/useRouter.js]: Mostly just out of curiosity - why add this permissiveness?

**Review by andreilee — APPROVED** [review]
Looks great!

File: nms/packages/magmalte/fbc_js_core/ui/hooks/useRouter.js (hunk 2 above)
Comment (thmsschmitt) [reply nms/packages/magmalte/fbc_js_core/ui/hooks/useRouter.js]: Just as safety net. We should still try to use it consistently until it gets removed by upgrading the react-router :)

File: nms/packages/magmalte/fbc_js_core/ui/stories/mui-components/appSideBar.stories.js (hunk 1 above)
Comment (thmsschmitt) [reply nms/packages/magmalte/fbc_js_core/ui/stories/mui-components/appSideBar.stories.js]: Yes, but it isn't running anymore

**Pull request merged**